src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -331,9 +392,22 @@ uct_dc_mlx5_ep_basic_init(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep)
 static UCS_F_ALWAYS_INLINE int
 uct_dc_mlx5_iface_dci_can_alloc(uct_dc_mlx5_iface_t *iface, uint8_t pool_index)
 {
+    ucs_assert_always(pool_index < iface->tx.num_dci_pools);
     return iface->tx.dci_pool[pool_index].stack_top < iface->tx.ndci;
 }
 
+static UCS_F_ALWAYS_INLINE int uct_dc_mlx5_iface_dci_ensure_pool_size(uct_dc_mlx5_iface_t *iface,
+                                                  uint8_t pool_index)
+{
+    ucs_status_t status = uct_dc_mlx5_dci_pool_add_dci(iface, pool_index,
```

Comment:
1. this function name is weird
2. a call to slow path function cannot be done before checking some condition, it's adding fast path overhead